### DISCUSSION TITLE
Livewire pipeline breaks on Laravel V10 because of empty return at the end of request pipeline

### DISCUSSION THREAD

**Discussion by program-mia** [livewire/livewire#5560] [7. Bugs (archived)] [answered]
Ayo!

**Issue:**
Livewire requests throwin 500 errors

**Setup:**
- Fresh Laravel 10 installation with only Livewire installed
- A route with "**guest**" middleware applied on it

**Description:**
Some users found out that Livewire breaks on V10 on subsequent requests. I just tested it out myself, and it seems to be the case. When trying to run a Livewire request through a route that has a "**guest**" middleware applied, it breaks, saying that the return type should be a Response, but null was returned.

The error happens because of this code from ``HttpConnectionHandler``:
```php
(new Pipeline(app()))
    ->send($request)
    ->through($filteredMiddleware)
    ->then(function() {
         // noop
    });
```

To recreate, you can just create a new project on V10, install Livewire and define a dummy method and a button with wire:click that triggers the method. Also add the "**guest**" middleware on the route that the Livewire component is in.

I found out that removing ``->then(...)`` solves the issue, BUT I am not sure why it was there in the first place. Is it safe to remove that empty return or is it something used later on by Livewire? I will probably check further later but wanted to bring this to light since maybe it's just as easy as removing the last ``->then(...)`` call from the code.

Cheers!

**Comment by joshhanley** [comment]
@program-mia thanks! It looks like it's because of the new return types that have been added in Laravel 10, where as the pipeline is returning null because of the noop.

<img width="1427" alt="image" src="https://user-images.githubusercontent.com/882837/219799283-be90a009-4448-4af5-9980-7a438f28495c.png">

 Not sure why that noop is there, we'll have to investigate that.

**Comment by joshhanley** [comment]
Have submitting a PR #5561 with a fix.

**Comment by joshhanley** [comment, marked as answer]
Fix for this has been merged in PR #5561 and tagged in v2.11.3.

**Reply by program-mia** [reply]
You're the best, Josh!

**Reply by joshhanley** [reply]
@program-mia no worries! 🙂 (not sure why someone has a confused reaction to that 😂)
